docs/about/contributing/modules.md

```diff
@@ -0,0 +1,449 @@
+
+## Module Integration
+
+```tf
+module "example" {
+  source   = "registry.coder.com/modules/example/coder"
+  version  = "1.0.0"
+  agent_id = coder_agent.main.id
+}
+```
+
+Browse available modules at [registry.coder.com](https://registry.coder.com).
+
+## Module Structure
+
+Every module consists of exactly four files:
+
+```text
+module-name/
+├── main.tf          # Terraform configuration with Coder resources
+├── main.test.ts     # Test suite
+├── README.md        # Documentation with frontmatter
+└── run.sh           # Installation script
+```
+
+### File Purposes
+
+- **`main.tf`**: Primary Terraform configuration defining Coder resources
+- **`main.test.ts`**: Test suite validating module functionality
+- **`README.md`**: Documentation with required frontmatter and usage examples
+- **`run.sh`**: Shell script for software installation and configuration
+
+## Understanding Coder Resources
+
+The Coder Terraform provider offers several resource types for different aspects of workspace functionality. Understanding when and how to use each resource is crucial for effective module development.
+
+### Coder Resources
+
+#### coder_script - Command Execution
+
+Execute commands during workspace lifecycle events. This is the primary mechanism for software installation, service configuration, and environment setup.
+
+**When to use**:
+
+- Installing software packages, binaries, or development tools
+- Configuring services and generating configuration files
+- Setting up directories, permissions, and initial workspace state
+- Running background services or daemons
+
+**Key properties**:
+
+- `agent_id`: The Coder agent to execute the script on
+- `display_name`: Name shown in the Coder dashboard during execution
+- `run_on_start`: Execute when workspace starts (most common)
+- `run_on_stop`: Execute during workspace shutdown for cleanup
+- `script`: The actual shell script content to execute
+- `start_blocks_login`: Whether script completion is required before user access
+
+Example:
+
+```tf
+resource "coder_script" "install" {
+  agent_id          = var.agent_id
+  display_name      = "Install Tool"
+  script            = templatefile("${path.module}/run.sh", {
+    VERSION = var.version
+    ARCH    = data.coder_provisioner.me.arch
+  })
+  run_on_start      = true
+  start_blocks_login = true
+}
+```
+
+#### coder_app - User Interface
+
+Create accessible applications in the Coder workspace interface, providing users with one-click access to tools and services.
+
+**When to use**:
+
+- Exposing web-based development tools (IDEs, dashboards)
+- Creating links to desktop applications via custom protocols
+- Providing access to running services
+
+**Key properties**:
+
+- `agent_id`: The Coder agent
+- `external`: `true` for protocol handlers, `false` for web apps
+- `healthcheck`: Monitor service availability
+- `subdomain`: Access method for web apps
+- `url`: Service URL or protocol handler
```

Comment:
`order` and `group` are also important to tell her. Maybe not if we link provider docs.